src/index.js

```diff
@@ -115,7 +116,10 @@ class KadDHT {
     /**
      * Random walk state, default true
      */
-    this.randomWalkEnabled = !options.hasOwnProperty('enabledDiscovery') ? true : Boolean(options.enabledDiscovery)
+    this.randomWalkEnabled = !options.randomWalk.hasOwnProperty('enabled') ? c.defaultRandomWalk.enabled : Boolean(options.randomWalk.enabled)
+    this.randomWalkQueriesPerPeriod = !options.randomWalk.hasOwnProperty('queriesPerPeriod') ? c.defaultRandomWalk.queriesPerPeriod : parseInt(options.randomWalk.queriesPerPeriod)
+    this.randomWalkInterval = !options.randomWalk.hasOwnProperty('interval') ? c.defaultRandomWalk.interval : parseInt(options.randomWalk.interval)
+    this.randomWalkTimeout = !options.randomWalk.hasOwnProperty('timeout') ? c.defaultRandomWalk.timeout : parseInt(options.randomWalk.timeout)
```

Comment:
I would recommend using [nodeutils/defaults-deep](https://github.com/nodeutils/defaults-deep#readme) in here, and in

```js
options.randomWalk = options.randomWalk || c.defaultRandomWalk
```

With the current code, if we receive a parameter as follows:

```
{
  enabled: true
}
```

The initial options object will only have that, as it exists. With my proposal, you can remove all the lines above here and default in the options above.